pkg/cmd/pipeline/crud.go

```diff
@@ -29,3 +33,25 @@ func IsDeployed(ctx context.Context, name, namespace string, c kubernetes.Interf
 	}
 	return cmap.Data[statusField] != ErrorStatus
 }
+
+// HasDeployedSomething checks if the pipeline has deployed any deployment/statefulset/job
```

Comment:
```suggestion
// HasDeployedSomething checks if the pipeline has deployed any deployment/statefulset
```

you are not checking jobs, not sure if we should or not

Reply:
We shouldn't because we don't support developing on jobs

Reply:
then I think we are coupling the logic of showing the CTA with the pipeline status. If this function returns a false, could be that something was deployed (a cronjob, a job, a daemonset or something else) successfully, but we are returning false, so we are marking the deploy as failed (check my previous comment)